### PR TITLE
Add cli option to specify src directory

### PR DESCRIPTION
**Pull request by nat-n**
In my project python packages live under the src directory. I want to be able to run the strawberry serve command from the root directory. The uvicorn cli [has a `--app-dir` option](https://github.com/encode/uvicorn/blob/master/uvicorn/main.py#L322) to support this use case. The present change adds the same mechanism to the strawberry CLI, also in the form of an `--app-dir` option.

## Checklist

- [x] My code follows the code style of this project.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/),so make sure to pick the appropriate type. If in doubt feel free to ask :)

**Comment by patrick91** [comment]
@nat-n looks good to me, should we follow uvicorn and name the parameter `--app-dir` as well?

**Comment by nat-n** [comment]
@patrick91 Thanks.

I was thinking the semantics are a slightly different. Uvicorn targets an ASGI _app_, whereas strawberry server targets a _schema_. Though consistency is also valuable; I don't feel strongly about it.

I'll prepare the rest of the PR.

File: RELEASE.md
Comment (patrick91) [inline RELEASE.md:3]: Let's add an example of using this here 😊
```suggestion
Add --app-dir CLI option to specify where to find the schema module to load when using the debug server
```

File: strawberry/cli/commands/server.py
Comment (patrick91) [inline strawberry/cli/commands/server.py:18]: @BryceBeagle what do you think of `--app-dir` vs `--src-dir`?

**Review by patrick91 — COMMENTED** [review]
Nice work so far, added a minor comment and asked @BryceBeagle for more feedback around the argument name 😊

File: strawberry/cli/commands/server.py
Comment (BryceBeagle) [reply strawberry/cli/commands/server.py:18]: I think `--src-dir` is more reflective of what the option actually does, but consistency is nice. I don't have a strong opinion on which is used.

Regardless, I think the `help` argument to the `@click.option` should provide a little note mentioning that this is the same functionality as the `--app-dir` flag that the uvicorn CLI provides.

Comment (nat-n) [reply strawberry/cli/commands/server.py:18]: @bryanculver The help message is copied almost verbatim from the uvicorn CLI. Are you suggesting to make it refer to the uvicorn CLI for context? It's already kinda verbose.

Screenshot 2021-04-05 at 18 24 15

**Comment by nat-n** [comment]
Updated the RELEASE.md with an example.

File: strawberry/cli/commands/server.py
Comment (BryceBeagle) [reply strawberry/cli/commands/server.py:18]: I assume you meant to tag me instead :stuck_out_tongue:. I was just suggesting adding a quick blurb that says something like "This is the same functionality as the `--app-dir` option that Uvicorn provides". I don't think it would be too verbose :slightly_smiling_face:

File: RELEASE.md
Comment (patrick91) [inline RELEASE.md:11]: 👍

File: strawberry/cli/commands/server.py
Comment (patrick91) [inline strawberry/cli/commands/server.py:23]:
```suggestion
    help=(
        "Look for the module in the specified directory, by adding this to the "
        "PYTHONPATH. Defaults to the current working directory. "
        "Works the same as `--app-dir` in uvicorn."
    ),
```

@BryceBeagle does this sound good?

**Review by patrick91 — COMMENTED** [review]
Nice work! Thanks for updating the PR with our suggestions. I think we can merge this after updating the help value :)

File: strawberry/cli/commands/server.py
Comment (BryceBeagle) [reply strawberry/cli/commands/server.py:23]: Looks perfect :+1:

**Review by patrick91 — APPROVED** [review]
@nat-n almost forgot, feel free to fill this form to get a sticker https://forms.gle/dmnfQUPoY5gZbVT67

And congrats on your first PR and release 😊

**Comment by nat-n** [comment]
@patrick91 Thanks! I like the way you've set things up here :)

**Pull request merged**
